### DISCUSSION TITLE
Styles source mystery

### DISCUSSION THREAD

**Discussion by afbagwell** [vladmandic/sdnext#2148] [Q&A]
I noted the change that styles are now  stored as individual JSON files in the models/styles folder.

To test this, I completely removed the styles.csv file in the main folder and restarted the app. I still have my styles. So far, so good. However if I try to manually rename, add or delete a style file through Windows Explorer, this is ignored and is not reflected in the drop-down after clicking the refresh button or restarting the server. In fact, the next time I refresh all the deleted styles return to the folder and drop-down.

Is this being worked on?

**Comment by vladmandic** [comment]
styles.csv -> models/styles/*.json is a one time operation. after that styles.csv is ignored.
refresh was just added.

**Comment by afbagwell** [comment]
Yeah I got that. All my styles were transferred over to the new folder. I moved my original styles.csv file for safe-keeping.

However I have gone into the models/styles folder and re-named several of the JSON files for clarity. Hitting refresh it does not display the new names. Restarting the app also does not display the names, and re-creates the files with original names so that now I have differently named dupes in the folder.

Example:
My Test Style.json
rename to MyTestStyle.json

refresh still shows "My Test Style" in the list.
restart/refresh continues to show "My Test Style" in the list, but now I have "My Test Style.json" and "MyTestStyle.json" in the folder.

**Reply by vladmandic** [reply]
a) anything thats found in styles.csv will be migrated to models/styles.
b) to rename, you need to change filename as well as name param inside the json.

and yes, its still work in progress.
